devices.go

```diff
@@ -155,8 +157,17 @@ func (d *deviceCache) IsDeviceLocked(id string) (exists, locked bool) {
 	return false, false
 }
 
-// RemoveById removes the specified (by id) device from the cache.
-func (d *deviceCache) RemoveById(id string) error {
+// RemoveByName removes the specified (by name) device from the cache.
+func (d *deviceCache) RemoveByName(name string) error {
+	err := svc.dc.DeleteByName(name)
+	if err != nil {
+		return err
+	}
+
+	if _, ok := d.devices[name]; ok {
+		delete(d.devices, name)
```

Comment:
Also should be removed from names map.

Reply:
Ok and I think Remove(dev *models.Device) is more appropriate